src/modules/location/index.ts

```diff
@@ -284,14 +284,16 @@ export class LocationModule {
    * @since 8.0.0
    */
   secondaryAddress(): string {
-    return this.faker.helpers
+    const format = this.faker.helpers
       .arrayElement(this.faker.definitions.location.secondary_address)
       .replace(/#+/g, (m) =>
         this.faker.string.numeric({
           length: m.length,
           allowLeadingZeros: false,
         })
       );
+
+    return this.faker.helpers.fake(format);
```

Comment:
IMO you can directly pass the template to the fake function.

```suggestion
    return this.faker.helpers
      .fake(this.faker.definitions.location.secondary_address)
      .replace(/#+/g, (m) =>
        this.faker.string.numeric({
          length: m.length,
          allowLeadingZeros: false,
        })
      );
```

Reply:
Thank you!